crypto/http/http_lib.c

```diff
@@ -59,44 +60,70 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
             if (pssl != NULL)
                 *pssl = 1;
             port = OSSL_HTTPS_PORT;
+            portnum = 443;
         } else if (strcmp(buf, OSSL_HTTP_NAME) != 0) {
             goto parse_err;
         }
 
-        /* Check for double slash */
+        /* check for double slash */
         if ((p[0] != '/') || (p[1] != '/'))
             goto parse_err;
         p += 2;
     }
     host = p;
 
-    /* Check for trailing part of path */
-    p = strchr(p, '/');
-    if (ppath != NULL && (*ppath = OPENSSL_strdup(p == NULL ? "/" : p)) == NULL)
-        goto err;
-    if (p != NULL)
-        *p = '\0'; /* Set start of path to 0 so hostname[:port] is valid */
-
-    p = host;
+    /* parse host name/address as far as needed here */
     if (host[0] == '[') {
-        /* ipv6 literal */
+        /* ipv6 literal, which may include ':' */
         host++;
-        p = strchr(host, ']');
-        if (p == NULL)
+        host_end = strchr(host, ']');
+        if (host_end == NULL)
             goto parse_err;
-        *p = '\0';
-        p++;
+        *host_end++ = '\0';
+    } else {
+        host_end = strchr(host, ':');
```

Comment:
Maybe some comments here
/* search for start of optional port */
...
/* search for start of optional path */
...

/* If there is no port or path so the remaining string is just the hostname */

Reply:
Done (using just more compact texts fitting in the line limit).